File: AutoIt.Foundation/AutoIt.Foundation.Common/Binding/Binding.cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;
using AutoIt.Foundation.Common.DataStruct.KeyValue;
using Compiler.Common;

namespace AutoIt.Foundation.Common.Binding
{
    /// <summary>
    /// 绑定
    /// </summary>
    public class Binding
    {
        /// <summary>
        /// 建立绑定
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="target">绑定目标</param>
        /// <param name="source">绑定源</param>
        public static void Bind<T>(Expression<Func<T>> target, Expression<Func<T>> source)
        {
            //将绑定信息添加到上下文中
            var group = Context.Current.GetOrSet(new List<BindInfo>());
            group.Add(new BindInfo(target, source));
        }

        /// <summary>
        /// 更新绑定目标
        /// </summary>
        public static void Update()
        {
            //从上下文中获取绑定信息
            var group = Context.Current.Get<List<BindInfo>>();

            //更新绑定目标
            if (group != null)
            {
                foreach (var item in group)
                {
                    item.Update();
                }
            }
        }
    }
}

File: AutoIt.Foundation/AutoIt.Foundation.Web.Scripts/Bind/Binding.ts
class Binding {
    //绑定信息
    static Key:string="Binding";

    //将源绑定到目标
    static Bind(target: FuncOne<any, any>, source: Func<any>) {
        var group = Context.Current();
        var bindGroup = <List<BindInfo>>group.Get(this.Key, new List<BindInfo>());

        bindGroup.Set(new BindInfo(target, source));
    }

    //更新所有绑定信息
    static Update() {
        var group = <List<BindInfo>>Context.Current().Get(this.Key);

        if (group != None) {
            group.ToEnumerble()
                .ForEach(item => {
                    item.Update();
                });
        }
    }
}